src/webServer/webServer.ts

```diff
@@ -137,6 +139,22 @@ namespace ts.server {
             /* eslint-enable no-restricted-globals */
 
             require: () => ({ module: undefined, error: new Error("Not implemented") }),
+            importServicePlugin: async (root: string, moduleName: string) => {
+                const packageRoot = combinePaths(root, "node_modules", moduleName);
+
+                const packageJsonResponse = await fetch(combinePaths(packageRoot, "package.json"));
+                const packageJson = await packageJsonResponse.json();
+                const browser = packageJson.browser;

```

Comment:
Do we only want to support `"browser"`, or do we think we'll need to support [NodeJS Conditional Exports](https://nodejs.org/dist/latest-v17.x/docs/api/packages.html#conditional-exports) such as the [community-defined `"browser"` condition](https://nodejs.org/dist/latest-v17.x/docs/api/packages.html#community-conditions-definitions)?

I'm fine with *not* supporting a `"browser"` export condition for the time being, as that would likely require either depending on our `node12`/`nodenext` module resolver or implementing NodeJS's conditional export logic if the `node12`/`nodenext` resolver isn't sufficient. Also, it seems fair enough for VS Code to require that async service plugins only uses the `"browser"` property.

Reply:
For VS Code extensions, we use a `main` field for desktop and a [`browser` field for web](https://code.visualstudio.com/api/extension-guides/web-extensions#web-extension-anatomy). I think a simple approach like that is fine here but we can align it more with the standards if you'd prefer

Reply:
I wouldn't worry about it for now, though we may revisit if we want to leverage this mechanism for other asynchronous imports later.